### PR TITLE
Feat: Add customizable Metrics Analysis to the rollout API 

### LINKED ISSUE #681: Add customizable Metrics Analysis to the rollout API
**What would you like to be added**:
Add customizable Metrics Analysis to the rollout API.
**Why is this needed**:
Flagger uses `MetricTemplate` to specify custom Metrics Analysis, which can give users more freedom for detection

### PR DESCRIPTION
**Pull request by Gidi233**
**What type of PR is this?**

/kind api-change
/kind feature

**What this PR does / why we need it**:
Providing users with customizable Metrics Analysis enables users to flexibly adjust according to different needs and scenarios, ensuring that users can obtain more accurate analysis results that meet specific business applications, thereby improving the operability of the system and the accuracy of data analysis.

**Which issue(s) this PR fixes**:
Fixes #681 

**Does this PR introduce a user-facing change?**:

```release-note
    ......
          trafficAnalysis:
             checkIntervalSeconds: 90
             checkFailedTimes: 2
             metrics:
             - name: request-success-rate
               intervalSeconds: 90
               thresholdRange:
                 min: 99
             - name: my-metric
               intervalSeconds: 90
               thresholdRange:
                 max: 99
               customMetric: 
                 provider: 
                   type: prometheus
                   address: http://flagger-prometheus.ingress-nginx:9090
                 query: |
                   sum(
                     rate(
                       http_requests_total{
                         status!~"5.*"
                       }[{{ interval }}]
                     )
                   )
                   /
                   sum(
                     rate(
                       http_requests_total[{{ interval }}]
                     )
                   ) * 100
             webhooks:
                 timeoutSeconds: 60
                 command:
                 - "hey -z 1m -q 10 -c 2 http://backend-canary.webapp:9898/"
    ......
```

### CODE REVIEW

File: docs/content/en/references/apps_v1alpha1_types.html [hunk 1]
```diff
@@ -1616,7 +1616,8 @@ <h3 id="apps.kurator.dev/v1alpha1.Metric">Metric
 </td>
 <td>
 <p>Name of the metric.
-Currently supported metric are <code>request-success-rate</code> and <code>request-duration</code>.</p>
+Currently supported metric are <code>request-success-rate</code> and <code>request-duration</code>.
+When Name is not <code>request-success-rate</code> or <code>request-duration</code>, you need to define the metric rule in CustomMetric.</p>
```
Comment (LiZhenCheng9527) [inline docs/content/en/references/apps_v1alpha1_types.html:1620]: Even if request-success-tare and request-duration are not available, you can use the metrics that come with the gateway without customising them.
Both scenarios should be made clear

**Review by LiZhenCheng9527 — COMMENTED** [review]
almost lgtm

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**